api4/license.go

```diff
@@ -296,3 +301,174 @@ func getPrevTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	license := c.App.Channels().License()
+	if license == nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.license.required", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if c.App.Cloud() != nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.not.allowed.for.cloud", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	// Customer Info & Usage Analytics
+	activeUserCount, err := c.App.Srv().Store().Status().GetTotalActiveUsersCount()
+	if err != nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.user.count.fail", nil, "", http.StatusInternalServerError)
+		return
+	}
+
+	// Webhook, calls, boards, and playbook counts
+	incomingWebhookCount, err := c.App.Srv().Store().Webhook().AnalyticsIncomingCount("")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.webhook.in.count.fail", nil, "", http.StatusInternalServerError)
+		return
+	}
+	outgoingWebhookCount, err := c.App.Srv().Store().Webhook().AnalyticsOutgoingCount("")
+	if err != nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.webhook.out.count.fail", nil, "", http.StatusInternalServerError)
+		return
+	}
+
+	// Plugin Data
+	trueUpReviewPlugins := model.TrueUpReviewPlugins{
+		ActivePluginNames:   []string{},
+		InactivePluginNames: []string{},
+	}
+
+	if pluginResponse, err := c.App.GetPlugins(); err == nil {
+		for _, plugin := range pluginResponse.Active {
+			trueUpReviewPlugins.ActivePluginNames = append(trueUpReviewPlugins.ActivePluginNames, plugin.Name)
+		}
+		trueUpReviewPlugins.TotalActivePlugins = len(trueUpReviewPlugins.ActivePluginNames)
+
+		for _, plugin := range pluginResponse.Inactive {
+			trueUpReviewPlugins.InactivePluginNames = append(trueUpReviewPlugins.InactivePluginNames, plugin.Name)
+		}
+		trueUpReviewPlugins.TotalInactivePlugins = len(trueUpReviewPlugins.InactivePluginNames)
+	}
+
+	// Authentication Features
+	config := c.App.Config()
+	mfaUsed := config.ServiceSettings.EnforceMultifactorAuthentication
+	ldapUsed := config.LdapSettings.Enable
+	samlUsed := config.SamlSettings.Enable
+	openIdUsed := config.OpenIdSettings.Enable
+	guessAccessAllowed := config.GuestAccountsSettings.Enable
+
+	authFeatures := map[string]*bool{
+		model.TrueUpReviewAuthFeaturesMfa:        mfaUsed,
+		model.TrueUpReviewAuthFeaturesADLdap:     ldapUsed,
+		model.TrueUpReviewAuthFeaturesSaml:       samlUsed,
+		model.TrueUpReviewAuthFeatureOpenId:      openIdUsed,
+		model.TrueUpReviewAuthFeatureGuestAccess: guessAccessAllowed,
+	}
+
+	authFeatureList := []string{}
+	for feature, used := range authFeatures {
+		if used != nil && *used {
+			authFeatureList = append(authFeatureList, feature)
+		}
+	}
+
+	reviewProfile := model.TrueUpReviewProfile{
+		ServerId:               c.App.TelemetryId(),
```

Comment:
username and password are things we wouldn't want to expose (and aren't exposed here). Personally identifiable information means sensitive or non-sensitive information that can be used to tie data to people in real life.

At most, I think this is something we would want to ask legal about. It does not seem like a security issue. Wanted to let John reply because the figma shows [some discussion](https://www.figma.com/file/1qWr4RtBjwQZyGQMT7z9Jp?node-id=503:38649#332756441) of consulting with legal already, and I don't want to duplicate conversation or cause confusion if this has already been discussed.
 
My concern is that we shouldn't overcollect information, and that doing so might lead to unfair levels of information or might lead to degraded likelihood of information being kept anonymous. My point is that we don't seem to need some of this information from a functionality perspective, and it seems like we are blurring functionality (is user count compliant with license seat limits) and data collection (various server stats).

One overcollection issue is that some of this information collection (talking about data other than the server id here, like specific plugin usage) isn't directly tied to license seat compliance, and that it could be used by us to infer how much value an organization perceives in its contract, and give Mattermost extra bargaining power that the customer didn't intend to disclose when they just thought they were complying with the license true-up process. At a minimum, that seems like something we would want to call out in out documents on privacy and data collection, if we do indeed want to collect the info.

Disclosing together the server's unique telemetry id and the organization's concrete information makes it so thereafter, any server-side tracking events it fires, we concretely know which organization is emitting them. That's something we don't presently know, and it makes that data that much less anonymous. It could also conceivably be used to gather server-specific usage data to unfairly inform us of the customer's perceived value of using Mattermost.

The other degraded privacy concern is that if the client-side tracking events also contain server telemetry id (to my knowledge, [they do not](https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/client/rudder.ts) at the present), or if there is an unanticipated future way to correlate client-side tracking events to a server telemetry id, now whoever knows that information knows concretely which organization any given client-tracking event belongs to. By not collecting server telemetry id in true up, we keep the id more anonymous, which limits how de-anonymized server and client tracking events can ever be.

Reply:
@neallred 

> are we sure we need/want to collect the server telemetry id?

Preference is to collect it, to keep it consistent with [instances that have telemetry turned on](https://mattermost.looker.com/dashboards/270?Account%20Owner=).

In the linked dashboard ^ , `Instance ID` is server telemetry ID, yes?

Re: the telemetry docs, I'll consult this with the docs team + legal.

On the Privacy Policy though,

> We refer to this information as “telemetry data,” and it does not include any end user personal identifiers or message contents.

This statement holds, as I don't think we collect "end user personal identifiers".

Reply:
I don't have access to view that dashboard, but my guess is that the Instance ID displayed there is the telemetry ID.

As follow-up and for transparency, we discussed in a team meeting that:
* We already collect license id together with telemetry id (https://github.com/mattermost/mattermost-server/blob/master/services/telemetry/telemetry.go#L855-L874) when telemetry is on
* If telemetry is off, telemetry events are not sent in the first place, so sending telemetry id and license info together reveal past telemetry data.
* As stated above, we will discuss more with docs/legal team about any changes needed to documentation, privacy policy, and/or data collection proposed in the ticket backing this PR.

Reply:
> I don't have access to view that dashboard, but my guess is that the Instance ID displayed there is the telemetry ID.

I've asked Daniel Sischy to grant you access

> If telemetry is off, telemetry events are not sent in the first place, so sending telemetry id and license info together reveal past telemetry data.

I'm not sure I understand. If telemetry events aren't sent in the first place, would it not reveal any past telemetry data?
@neallred

Reply:
I belive telemetry can be enabled and disabled as much as the customer likes, so they may have had it one in the past but now want it disabled @johndavidlugtu